funnel/assets/js/account_form.js

```diff
@@ -0,0 +1,187 @@
+import 'htmx.org';
+import Form from './utils/formhelper';
+
+window.Hasgeek.Accountform = ({
+  formId,
+  usernameField,
+  passwordField,
+  passwordForm,
+  accountUsernameUrl,
+  passwordCheckUrl,
+}) => {
+  $(formId)
+    .parsley()
+    .subscribe('parsley:field:validated', () => {
+      if ($(formId).parsley().isValid())
+        $(formId).addClass('parsley-valid').removeClass('parsley-invalid');
+      else $(formId).addClass('parsley-invalid').removeClass('parsley-valid');
+    });
+
+  let typingTimer;
+  let typingTimerUsername;
+  const typingWaitInterval = 100;
+  let waitingForResponse = false;
+  let waitingForUsernameResponse = false;
+
+  async function checkUsernameAvailability(field) {
+    if (!waitingForUsernameResponse && $(field).val()) {
+      waitingForUsernameResponse = true;
+      const response = await fetch(accountUsernameUrl, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/x-www-form-urlencoded',
+          'X-Requested-With': 'XMLHttpRequest',
```

Comment:
Can we drop the XHR header and change to JSON submit in this PR?

Reply:
Changed